### PR TITLE
chore: update README with `docker-compose`

### PR DESCRIPTION
**Pull request by hash070**
Add `docker-compose.yml` to make it easy to deploy with docker compose and simple to migrate all container data.

### CODE REVIEW

**Comment by boojack** [comment]
There already have a sample docker compose yaml file in `./quickstart/docker-compose.yaml`. Maybe you could update it please.

**Comment by hash070** [comment]
Oh, I don't know if the compose file is in there because you didn't mention it in readme.
This file is usually placed in the root directory, so I moved it and updated the `README.md`.

**Comment by boojack** [comment]
> Oh, I don't know if the compose file is in there because you didn't mention it in readme. This file is usually placed in the root directory, so I moved it and updated the `README.md`.

Can I get the reason about why this file must be in the root directory? I think it's just an example code about how to use docker compose starting memos, and the user maybe don't want to clone the repo firstly.

**Comment by hash070** [comment]
> > Oh, I don't know if the compose file is in there because you didn't mention it in readme. This file is usually placed in the root directory, so I moved it and updated the `README.md`.
> 
> Can I get the reason about why this file must be in the root directory? I think it's just an example code about how to use docker compose starting memos, and the user maybe don't want to clone the repo firstly.
> 
> In my option, the `./quickstart` folder will be more suitable.

Well, in my opinion, putting the docker compose file in the root directory will make it easier to spot, which could avoid someone who doesn't know this file exists ( just like me >_<
What's more, the user who cloned this repo can use the docker compose command at once after docker build action.
(And finally, putting this single file in a standalone dictionary seems to be meaningless.

But after all, the decision is yours. I like your project, and I just want to make this project better. Thanks for your contribution to the open source community.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,23 @@ docker run -d --name memos -p 5230:5230 -v ~/.memos/:/var/opt/memos neosmemo/mem
 
 Memos should be running at [http://localhost:5230](http://localhost:5230). If the `~/.memos/` does not have a `memos_prod.db` file, then memos will auto generate it.
 
+## Deploy with Docker Compose
```
Comment (boojack) [inline README.md:35]:
```suggestion
### With `docker-compose`
```

File: README.md [hunk 2]
```diff
@@ -32,6 +32,23 @@ docker run -d --name memos -p 5230:5230 -v ~/.memos/:/var/opt/memos neosmemo/mem
...
+mkdir -p ~/docker/memos && cd ~/docker/memos
+
+# Get the Docker compose file and you can modify it to suit your needs. 
+wget https://raw.githubusercontent.com/usememos/memos/main/quickstart/docker-compose.yaml
+
+# Run the container
+docker-compose up -d
+```
+
+It should be also running at [http://localhost:5230](http://localhost:5230). If there is no `memos_prod.db` file in `~/docker/memos/data`, then memos will auto generate it.
```
Comment (boojack) [inline README.md:50]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -32,6 +32,23 @@ docker run -d --name memos -p 5230:5230 -v ~/.memos/:/var/opt/memos neosmemo/mem
 
 Memos should be running at [http://localhost:5230](http://localhost:5230). If the `~/.memos/` does not have a `memos_prod.db` file, then memos will auto generate it.
 
+## Deploy with Docker Compose
+
+The advantages of using Docker Compose to deploy are that you don't have to remember the start up command or the relationship of the containers, and it's simple to migrate all of your containers' data.
```
Comment (boojack) [inline README.md:37]: I think README should catch the point about how to run with `docker-compose`. So I'd like to write the full description in the future docs.

Comment (boojack) [inline README.md:37]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -32,6 +32,23 @@ docker run -d --name memos -p 5230:5230 -v ~/.memos/:/var/opt/memos neosmemo/mem
 
 Memos should be running at [http://localhost:5230](http://localhost:5230). If the `~/.memos/` does not have a `memos_prod.db` file, then memos will auto generate it.
 
+## Deploy with Docker Compose
+
+The advantages of using Docker Compose to deploy are that you don't have to remember the start up command or the relationship of the containers, and it's simple to migrate all of your containers' data. 
+
+```bash
+# Make and enter the directory for the container
+mkdir -p ~/docker/memos && cd ~/docker/memos
```
Comment (boojack) [inline README.md:41]: Be consistency with previous folder will be better?
```suggestion
mkdir -p ~/.memos && cd ~/.memos
```

**Comment by boojack** [comment]
> But after all, the decision is yours. I like your project, and I just want to make this project better. Thanks for your contribution to the open source community. 

Thanks for your interesting, let's make the project even better!

---

> putting this single file in a standalone dictionary seems to be meaningless.

I think the `docker-compose` file **is not a source code file but a exmaple code**. If the user knows a little about docker, then he should know that docker images can be running with `docker-compose`. If the user will not use this file by cloning repo but a url, I'd like to keep root directory clean. WDYT?

**Comment by boojack** [comment]
FYI, updated docker compose file sample in README: https://github.com/usememos/memos#docker-compose

**Pull request closed without merging**